src/pages/user/viewer/details.jsx

```diff
@@ -1,36 +1,107 @@
+        margin-bottom: 1rem;
+        display: flex;
+
+        @media screen and (max-width: 480px) {
+            width: 100%;
+        }
     `,
 }
 
 export const UserViewerDetails = ({ user }) => {
+
+    let history = useHistory();
+
+    // Import the following React contexts:
+    const loggedinUser = useContext(AuthenticationContext);
+
+    // Function availibility control:
+    let activationStateButtonAvailibility;
+    let modifyUserStateButtonAvailibility;
+    let deleteAvatarButtonAvailibility;
+    let printCrewCardButtonAvailibility;
+
     const [ membershipState, setMembershipState ] = useState(null);
-    const [ activationState, setActivationState] = useState(null);
-    const [ loading, setLoading ] = useState(false);
+    const [ activationState, setActivationState ] = useState(null);
+
+    const [ loading, setLoading ] = useState(true);
+
+    // Check if user has "admin" role and make the following functions available:
+    if (loggedinUser.roles.includes("admin")) {
+        activationStateButtonAvailibility = true;
+        modifyUserStateButtonAvailibility = true;
+    }
+
+    // Avatar button availability logic, check if the user is him/herself or is admin or hr_admin, and check if the user has an avatar to make the button available:
+    if (user.avatar_uuid) {
+        printCrewCardButtonAvailibility = true;
+        if((loggedinUser.roles.includes("admin") || loggedinUser.roles.includes("hr_admin")) || loggedinUser.authUser.uuid === user.uuid) {
+            deleteAvatarButtonAvailibility = true;
+        }
+    }
 
     const reload = async () => {
         setLoading(true);
-        const [ membershipState, activationState ] = await Promise.all([
-            User.getUserMembershipStatus(user.uuid),
-            User.getUserActivationState(user.uuid),
-        ])
-        setMembershipState(membershipState);
-        setActivationState(activationState);
+
+        // Try to get user information and set the information as states which can be used later or throw an error.
+        try {
+            const [ activationState, membershipState ] = await Promise.all([
+                User.getUserActivationState(user.uuid),
+                User.getUserMembershipStatus(user.uuid)
+            ])
+            setActivationState(activationState);
+            setMembershipState(membershipState);
+        } catch(e) {
+            console.error("An error occured while attempting to gather user information:\n" + e);
+        }
+
+        // Logic finished, show the user details page:
         setLoading(false);
     }
 
+    // Function to activate the user
+    const activateUser = async () => {
+        if(window.confirm("Er du sikker på at du vil aktivere denne brukeren?")) {
+            try { 
+                await User.activateUser(user.uuid);
+                reload();
+            } catch(e) {
+                console.error("An error occured while attempting to update the user.\n" + e);
```

Comment:
Same here. Don't catch if you aren't showing the user.

In general you should only catch exceptions if you plan to recover from them.